src/stieltjes.jl

```diff
@@ -75,4 +74,173 @@ end
     D = T \ apply(*, log.(abs.(x .- x')), P)
     c = inv(2*kr.A)
     T[kr,:] * Diagonal(Vcat(2*convert(V,π)*c*log(c), 2c*D.diag.args[2]))
-end
\ No newline at end of file
+end
+
+#################################################
+# ∫f(x)g(x)(t-x)^a dx evaluation where f and g given in coefficients
+#################################################
+# recognize structure of W = P' * ((t .- x).^a .* P)
+const PowKernelIntegralPoint{T, V, D, F} = QuasiArrays.ApplyQuasiArray{T, 2, typeof(*), Tuple{QuasiArrays.QuasiAdjoint{T, Legendre{Float64}}, QuasiArrays.BroadcastQuasiArray{T, 2, typeof(*), Tuple{QuasiArrays.BroadcastQuasiArray{T, 1, typeof(^), Tuple{ContinuumArrays.AffineQuasiVector{T, V, Inclusion{V, D}, T}, F}}, Legendre{Float64}}}}} # method only works for Legendre for now, so require Legendre OPs.
+
+####
+# methods
+####
+# if needed repeatedly due to caching it's generally faster to store PowerLawIntegral directly and use that
+function dot(f::AbstractVector{T}, K::PowKernelIntegralPoint, g::AbstractVector{T}) where T
+    # generate operator
+    a = K.args[2].args[1].args[2]
+    t = K.args[2].args[1].args[1][0.]
+    PP = K.args[2].args[2]
+    K = PowerLawIntegral(PP,BigFloat("$a"),BigFloat("$t"))
+    # check if infinite
+    (lf, lg) = (length(f),length(g))
+    maxl = maximum((lf, lg))
+    (maxl<∞) && return dot(pad(f,maxl),K[1:maxl,1:maxl],pad(g,maxl))
+    ((lf<∞) || (lg<∞)) && error("TODO: Currently only both finite or both infinite.")
+    (i,conv1,conv2) = (0,1,2)
+    while abs(conv1-conv2)>1e-15
+        i = i+1
+        conv1 = dot(f[1:i*20],K[1:i*20,1:i*20],g[1:i*20])
+        conv2 = dot(f[1:2*i*20],K[1:2*i*20,1:2*i*20],g[1:2*i*20])
+    end
+    return conv2
+end
+function *(g::Adjoint, K::PowKernelIntegralPoint, f::AbstractVector)
+    return dot(g',K,f)
+end
+
+####
+# cached operator implementation
+####
+# Constructors support BigFloat and it's recommended to use them for high orders.
+mutable struct PowerLawIntegral{T, PP<:AbstractQuasiMatrix} <: AbstractCachedMatrix{T}
```

Comment:
This is a bad name: there is no `Integral`. It's literally just multiplication, that is, this represents the exact same matrix as `(t*I - X)^a` if `X` is the jacobi matrix.

Reply:
This isn't quite true as it's currently implemented but we can easily make it so that it is!

Put simply, the matrix that we are generating here is already P'P * PowKernelMatrix(...) in your terminology, so it isn't PowKernelMatrix(...) itself and it gives a different result from (t*I - X)^a (it's that but scaled by Legendre orthogonality coefficients). So that's why I called it PowerLawIntegral, because it does already contain the knowledge that it's there for integration.

We have a lot of ways to implement this differently but the recurrence method I derived much more naturally produces P'P * PowKernelMatrix(...). If you want PowKernelMatrix itself then the most simple way to get it is to compute PowerLawIntegral as it currently stands and then remove the scaling by multiplying with a diagonal matrix.

This seems in line with your other comment about wanting `P' * ((t .- x) .^ a .* P) == P'P * PowKernelMatrix(...)`, so I guess I'll just change it to that?

Reply:
Does this just mean we want to focus on `Normalized(Legendre()`? So that `P'P == I`?

Reply:
If P'P=I, then (t*I - X)^a is symmetric and indeed PowerLawIntegral = PowerLawMatrix. I haven't done it but am fairly certain I can modify the recurrence to be valid for normalized Legendre instead. Then there would be no need to have two types of operators here. It's up to - if you'd prefer that I can do it.

Reply:
I think it will be a lot cleaner.

We may wish to improve `Normalized(Legendre())` to use explicit formulae for the Jacobi matrix instead of the generic fall back (which is to symmetrize the Jacobi operator)

Reply:
That's just n/sqrt((4*n^2-1)) on the sub and super diagonal, right? Should I overload it for normalized Legendre in normalized.jl?

Reply:
Since normalized.jl is loaded first it should be in jacobi.jl. (Or if you feel like it you could move out the `Legendre()` code to a new file, legendre.jl). Something like this would probably be the simplest:
```julia
function jacobimatrix(::Normalized{<:Any,<:Legendre})
   b = ...#  super-diagonal as lazy BroadcastVector. Note first entry is ignored
    Symmetric(_BandedMatrix(b', (-1,1)))
end
recurrencecoefficients(:: Normalized{<:Any,<:Legendre}) = ...
```
(Unfortunately we can't use `SymTridiagonal(Zeros(∞), b)` since `SymTridiagonal` requires both vectors to be the same type. Down the line I may make my own more general version of `SymTridiagonal`.)

Reply:
> That's just n/sqrt(4*n^2-1) on the sub and super diagonal, right? 

No idea but easy to check for yourself by making sure you match current behaviour: 
```julia
julia> jacobimatrix(Normalized(Legendre()))
∞×∞ Symmetric{Float64,BandedMatrices.BandedMatrix{Float64,LazyArrays.ApplyArray{Float64,2,typeof(vcat),Tuple{Adjoint{Float64,FillArrays.Zeros{Float64,1,Tuple{InfiniteArrays.OneToInf{Int64}}}},Adjoint{Float64,LazyArrays.BroadcastArray{Float64,1,typeof(/),Tuple{LazyArrays.BroadcastArray{Float64,1,typeof(*),Tuple{LazyArrays.BroadcastArray{Float64,1,typeof(/),Tuple{InfiniteArrays.InfStepRange{Float64,Float64},InfiniteArrays.InfStepRange{Int64,Int64}}},ClassicalOrthogonalPolynomials.NormalizationConstant{Float64,Legendre{Float64}}}},SubArray{Float64,1,ClassicalOrthogonalPolynomials.NormalizationConstant{Float64,Legendre{Float64}},Tuple{InfiniteArrays.InfUnitRange{Int64}},false}}}}}},InfiniteArrays.OneToInf{Int64}}} with indices OneToInf()×OneToInf():
 0.0      0.57735    ⋅         ⋅         ⋅         ⋅         ⋅         ⋅         ⋅         ⋅         ⋅         ⋅         ⋅        …  
 0.57735  0.0       0.516398   ⋅         ⋅         ⋅         ⋅         ⋅         ⋅         ⋅         ⋅         ⋅         ⋅           
  ⋅       0.516398  0.0       0.507093   ⋅         ⋅         ⋅         ⋅         ⋅         ⋅         ⋅         ⋅         ⋅           
  ⋅        ⋅        0.507093  0.0       0.503953   ⋅         ⋅         ⋅         ⋅         ⋅         ⋅         ⋅         ⋅           
  ⋅        ⋅         ⋅        0.503953  0.0       0.502519   ⋅         ⋅         ⋅         ⋅         ⋅         ⋅         ⋅           
  ⋅        ⋅         ⋅         ⋅        0.502519  0.0       0.501745   ⋅         ⋅         ⋅         ⋅         ⋅         ⋅        …  
  ⋅        ⋅         ⋅         ⋅         ⋅        0.501745  0.0       0.50128    ⋅         ⋅         ⋅         ⋅         ⋅           
  ⋅        ⋅         ⋅         ⋅         ⋅         ⋅        0.50128   0.0       0.500979   ⋅         ⋅         ⋅         ⋅           
  ⋅        ⋅         ⋅         ⋅         ⋅         ⋅         ⋅        0.500979  0.0       0.500773   ⋅         ⋅         ⋅           
  ⋅        ⋅         ⋅         ⋅         ⋅         ⋅         ⋅         ⋅        0.500773  0.0       0.500626   ⋅         ⋅           
  ⋅        ⋅         ⋅         ⋅         ⋅         ⋅         ⋅         ⋅         ⋅        0.500626  0.0       0.500517   ⋅        …  
  ⋅        ⋅         ⋅         ⋅         ⋅         ⋅         ⋅         ⋅         ⋅         ⋅        0.500517  0.0       0.500435     
  ⋅        ⋅         ⋅         ⋅         ⋅         ⋅         ⋅         ⋅         ⋅         ⋅         ⋅        0.500435  0.0          
  ⋅        ⋅         ⋅         ⋅         ⋅         ⋅         ⋅         ⋅         ⋅         ⋅         ⋅         ⋅        0.50037      
  ⋅        ⋅         ⋅         ⋅         ⋅         ⋅         ⋅         ⋅         ⋅         ⋅         ⋅         ⋅         ⋅           
 ⋮                                                ⋮                                                 ⋮                             ⋱  
```

Reply:
Okay, yes that checks out. I'll add that in a first step and then modify the PowerLawMatrix stuff to be for normalized Legendre polynomials instead.

Reply:
@dlfivefifty The recurrence seems to be unstable when used to directly generate the normalized Legendre version, it degenerates after only a couple dozen orders, whereas it's perfectly stable into thousands and higher if we do Diagonal() * unnormalized(). Not exactly sure why that is, but it doesn't look like there's an obvious way around it.

Reply:
that's very weird..

Reply:
I thought so too. I guess I'll keep looking for a bug that could be causing this but it certainly would be a strange bug to be accurate to order 20-30 and then become unstable when I'm basically just modifying the recurrence with the normalizations.

Reply:
Okay I figured it out but basically by luck. There was a strange interaction caused by a call to BigFloat in a wrong place. Commit with normalized version coming up.